File: sources/ptm/contents/code/tooltip.py
# -*- coding: utf-8 -*-

############################################################################
#   This file is part of pytextmonitor                                     #
#                                                                          #
#   pytextmonitor is free software: you can redistribute it and/or         #
#   modify it under the terms of the GNU General Public License as         #
#   published by the Free Software Foundation, either version 3 of the     #
#   License, or (at your option) any later version.                        #
#                                                                          #
#   pytextmonitor is distributed in the hope that it will be useful,       #
#   but WITHOUT ANY WARRANTY; without even the implied warranty of         #
#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
#   GNU General Public License for more details.                           #
#                                                                          #
#   You should have received a copy of the GNU General Public License      #
#   along with pytextmonitor. If not, see http://www.gnu.org/licenses/     #
############################################################################

from PyQt4.QtCore import *
from PyQt4.QtGui import *



class Tooltip():
    def __init__(self, parent):
        """class definition"""
        self.parent = parent
    
    
    def addValue(self, type, value=0.0, tooltipNum=100):
        """function to add value to list"""
        if (len(self.parent.tooltipValues[type]) > tooltipNum):
            self.parent.tooltipValues[type] = self.parent.tooltipValues[type][1:]
        self.parent.tooltipValues[type].append(value)
    
    
    def createGraphic(self, types, colors, bounds, values, widget):
        """function to create graph"""
        widget.clear()
        maxOne = [100.0, 100.0]
        pen = QPen()
        bounds['down'] = 1.0
        for value in values['down']:
            if (bounds['down'] < value):
                bounds['down'] = value
        for value in values['up']:
            if (bounds['down'] < value):
                bounds['down'] = value
        bounds['up'] = bounds['down']
        down = False
        for type in types:
            norm = [maxOne[0] / len(values[type]), maxOne[1] / (1.5 * bounds[type])]
            pen.setColor(QColor(colors[type]))
            if (down):
                shift = (types.index(type) - 1) * maxOne[0]
            else:
                shift = types.index(type) * maxOne[0]
            for i in range(len(values[type])-1):
                x1 = i * norm[0] + shift
                y1 = -values[type][i] * norm[1]
                x2 = (i + 1) * norm[0] + shift
                y2 = -values[type][i+1] * norm[1]
                widget.addLine(x1, y1, x2, y2, pen)
            if (type == 'down'):
                down = True
    

File: sources/awesome-widget/tooltip.cpp
/***************************************************************************
 *   This file is part of pytextmonitor                                    *
 *                                                                         *
 *   pytextmonitor is free software: you can redistribute it and/or        *
 *   modify it under the terms of the GNU General Public License as        *
 *   published by the Free Software Foundation, either version 3 of the    *
 *   License, or (at your option) any later version.                       *
 *                                                                         *
 *   pytextmonitor is distributed in the hope that it will be useful,      *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
 *   GNU General Public License for more details.                          *
 *                                                                         *
 *   You should have received a copy of the GNU General Public License     *
 *   along with pytextmonitor. If not, see http://www.gnu.org/licenses/    *
 ***************************************************************************/

#include "awesome-widget.h"

#include <pdebug/pdebug.h>


void AwesomeWidget::initValues()
{
    if (debug) qDebug() << PDEBUG;
}


void AwesomeWidget::reinit()
{
    if (debug) qDebug() << PDEBUG;

    mainLayout = new QGraphicsLinearLayout();
    mainLayout->setContentsMargins(1, 1, 1, 1);
    setLayout(mainLayout);

    if (configuration[QString("background")].toInt() == 0)
        setBackgroundHints(NoBackground);
    else
        setBackgroundHints(DefaultBackground);
    if (configuration[QString("layout")].toInt() == 0)
        mainLayout->setOrientation(Qt::Horizontal);
    else
        mainLayout->setOrientation(Qt::Vertical);
    if (configuration[QString("leftStretch")].toInt() == 2)
        mainLayout->addStretch(1);
    if (configuration[QString("popup")].toInt() == 0)
        textLabel->setPopupEnabled(true);
    mainLayout->addItem(textLabel);
    if (configuration[QString("rightStretch")].toInt() == 2)
        mainLayout->addStretch(1);

    keys = getKeys();
    foundKeys = findKeys();
    initValues();
    values[QString("netdev")] = getNetworkDevice();
//    thread()->wait(60000);
    connectToEngine();
    resize(10, 10);
}


void AwesomeWidget::updateText()
{
    if (debug) qDebug() << PDEBUG;

    QString text = configuration[QString("text")];
    for (int i=0; i<foundKeys.count(); i++)
        text.replace(QString("$") + foundKeys[i] + QString("$"), values[foundKeys[i]]);
    textLabel->setText(text);
}


void AwesomeWidget::updateTooltip()
{
    if (debug) qDebug() << PDEBUG;
}
